src/lib/multicall/useMulticall.ts

```diff
@@ -50,14 +51,19 @@ export function useMulticall<TConfig extends MulticallRequestConfig<any>, TResul
       try {
         // prettier-ignore
         const request = typeof params.request === "function"
-            ? params.request(chainId, params.key as CacheKey)
+            ? await params.request(chainId, params.key as CacheKey)
             : params.request;
 
         if (Object.keys(request).length === 0) {
           throw new Error(`Multicall request is empty`);
         }
 
-        const responseOrFailure = await executeMulticall(chainId, signer, request);
+        let responseOrFailure: any;
+        if (params.inWorker) {
+          responseOrFailure = await executeMulticallWorker(chainId, request);
```

Comment:
in case smth happens to a worker (not loaded for any reason) would be nice to fallback to `executeMulticall` here

Reply:
agree, done